app/Http/Controllers/Api/RequestApiController.php

```diff
@@ -42,23 +44,31 @@ public function showSingleRequest(int $requestId): JsonResponse
      * Create a new request for the endpoint.
      * Dispatch ExecuteRequestJob.
      *
-     * @param Request $request
+     * @param WebRequest $webRequest
      * @return Response
      */
-    public function storeRequest(Request $request): Response
+    public function storeRequest(WebRequest $webRequest): Response
     {
-        $this->validate($request, [
+        $this->validate($webRequest, [
             'url' => 'required|url',
             'method' => 'required|in:'.implode(',', \App\Request::getAllowedMethods()),
+            'request_headers' => 'array',
+            'request_headers.*.name' => 'required',
+            'request_headers.*.value' => 'required',
         ]);
 
-        $endpoint = \App\Endpoint::firstOrCreate($request->only('url', 'method'));
-        $endpointRequest = $endpoint->requests()->create();
+        /* @var Endpoint $endpoint */
+        $endpoint = \App\Endpoint::firstOrCreate($webRequest->only('url', 'method'));
 
-        $this->dispatch(new ExecuteRequestJob($endpointRequest));
+        /* @var Request $request */
+        $request = $endpoint->requests()->create();
+
+        $request->requestHeaders()->createMany($webRequest->get('request_headers', []));
```

Comment:
Empty `[]`?

Reply:
Empty [] is the default value for request_headers when not present in the request.